frame/support/src/metadata.rs

```diff
@@ -414,6 +418,37 @@ mod tests {
 	#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 	pub struct TestRuntime;
 
+	impl crate::traits::PalletInfo for TestRuntime {
+		fn index<P: 'static>() -> Option<usize> {
+			let type_id = sp_std::any::TypeId::of::<P>();
+			if type_id == sp_std::any::TypeId::of::<system::Module<TestRuntime>>() {
+				return Some(0)
+			}
+			if type_id == sp_std::any::TypeId::of::<EventModule>() {
+				return Some(1)
+			}
+			if type_id == sp_std::any::TypeId::of::<EventModule2>() {
+				return Some(2)
+			}
+
+			None
+		}
+		fn name<P: 'static>() -> Option<&'static str> {
+			let type_id = sp_std::any::TypeId::of::<P>();
+			if type_id == sp_std::any::TypeId::of::<system::Module<TestRuntime>>() {
+				return Some("System")
+			}
+			if type_id == sp_std::any::TypeId::of::<EventModule>() {
+				return Some("EventModule")
+			}
+			if type_id == sp_std::any::TypeId::of::<EventModule2>() {
+				return Some("EventModule2")
+			}
+
+			None
+		}
+	}
```

Comment:
Note: I didn't use construct_runtime here because I don't think it is really needed for the test, so the easiest is just a manual implementation.